### PR TITLE
MySQL 4byte support

### PR DESCRIPTION
**Pull request by nickvergessen**
- [x] Requires https://github.com/nextcloud/activity/pull/53

Downstream https://github.com/owncloud/core/pull/17978

@MorrisJobke can you setup drone with 4byte?

### CODE REVIEW

**Comment by MorrisJobke** [comment]
> @MorrisJobke can you setup drone with 4byte?

Yes.

File: .drone.yml [hunk 1]
```diff
@@ -323,50 +331,7 @@ pipeline:
...
-      PHP: 7.1
-    - DB: sqlite
-      PHP: 5.6
-    - DB: sqlite
-      PHP: 7.0
-    - DB: sqlite
-      PHP: 7.1
-    - DB: mysql
-      PHP: 5.6
-    - DB: postgres
```
Comment (MorrisJobke) [inline .drone.yml]: Because I currently can't test locally I removed this to have faster feedback ;)

**Comment by MorrisJobke** [comment]
I rebased to squash the testing commits

**Comment by MorrisJobke** [comment]
@nickvergessen Your turn now :)

**Comment by nickvergessen** [comment]
Current problem:
Update with activity app fails. We try to change the collation `Repair info: Change collation for oc_activity` before the database is updated.
This results in:
```
Syntax error or access violation: 1118 Row size too large. The maximum row size for the used table type, not counting BLOBs, is 65535. This includes storage overhead, check the manual. You have to change some columns to TEXT or BLOBs
```

Which is what would be solved by https://github.com/nextcloud/activity/pull/53 
But the repair step is executed before the update. My suggestion would be to ignore this error in the before step and retry it after the update was completed.

I commited that

**Comment by MorrisJobke** [comment]
```
1) Test\Repair\RepairCollationTest::testCollationConvert
Argument 2 passed to OC\Repair\Collation::__construct() must implement interface OCP\ILogger, instance of OC\DB\Connection given, called in /drone/src/github.com/nextcloud/server/tests/lib/Repair/RepairCollationTest.php on line 66 and defined
/drone/src/github.com/nextcloud/server/lib/private/Repair/Collation.php:54
/drone/src/github.com/nextcloud/server/tests/lib/Repair/RepairCollationTest.php:66
```

**Comment by MorrisJobke** [comment]
```
3) Test\Files\Cache\CacheTest::testFolder with data set #2 ('😐, NEUTRAL_FACE, UTF-8 hex F09F9890')
MySQL doesn't support 4 byte UTF-8
/drone/src/github.com/nextcloud/server/tests/lib/Files/Cache/CacheTest.php:117
4) Test\Files\Cache\MoveFromCacheTraitTest::testFolder with data set #2 ('😐, NEUTRAL_FACE, UTF-8 hex F09F9890')
MySQL doesn't support 4 byte UTF-8
/drone/src/github.com/nextcloud/server/tests/lib/Files/Cache/CacheTest.php:117
5) Test\Files\Cache\Wrapper\CacheJailTest::testFolder with data set #2 ('😐, NEUTRAL_FACE, UTF-8 hex F09F9890')
MySQL doesn't support 4 byte UTF-8
/drone/src/github.com/nextcloud/server/tests/lib/Files/Cache/CacheTest.php:117
6) Test\Files\Cache\Wrapper\CachePermissionsMaskTest::testFolder with data set #2 ('😐, NEUTRAL_FACE, UTF-8 hex F09F9890')
MySQL doesn't support 4 byte UTF-8
/drone/src/github.com/nextcloud/server/tests/lib/Files/Cache/CacheTest.php:117
```

**Comment by MorrisJobke** [comment]
Rebased for CI

**Comment by nickvergessen** [comment]
Test pass, ready to go.

Second review @rullzer @LukasReschke

**Comment by MorrisJobke** [comment]
:+1:

**Comment by LukasReschke** [comment]
LGTM

**Pull request merged**